### ISSUE TITLE
[Bug] Fail to use self-deployed open source model when upgrading

### ISSUE THREAD

**Issue by sunxichen** [BoundaryML/baml#983]
## Issue Description

After upgrading from Baml v0.53.1 to v0.55.3, I encountered an issue with the request body format for non-multimodal models. This affects both the `openai` as client provider and the `openai-generic` as a client provider.

### Current Behavior

In v0.55.3, the content field in the request body when calling /v1/chat/completions endpoint is structured as follows:

```json
{
    "model": "qwen2-72b-instruct",
    "messages": [
        {
            "role": "system",
            "content": [
                {
                    "type": "text",
                    "text": "Your goal is to ..."
                }
            ]
        }
        // other messages
    ]
}
```

### Previous Behavior (v0.53.1)

In v0.53.1, the content field was a simple string:

```json
{
    "model": "qwen2-72b-instruct",
    "messages": [
        {
            "role": "system",
            "content": "Your goal is to ..."
        }
        // other messages
    ]
}
```
### Problem

0.55.3 format works fine for multimodal models like GPT-4o or Qwen2-VL for open-sourced vl model. However, for text-only models like Qwen2-72B-instruct, it causes errors. Many self-hosted LLM frameworks (e.g., Xinference or vLLM) do not support the "type" and "text" fields in their `/v1/chat/completions` API for text-only models.

Maybe we could add an options field in `clients.baml` when declare a client to distinguish between multimodal and non-multimodal models? This would allow Baml to use different input formats based on the model type:

1. For multimodal models: Use the new format with "type" and "text" fields.
2. For text-only models: Use the original format with a simple string content.

**Comment by aaronvg** [comment]
Thanks for this report! We will see how we can patch this tomorrow!

**Comment by aaronvg** [comment]
we have a fix nearly ready -- it will go out tonight

**Comment by aaronvg** [comment]
This is now fixed in 0.57.0. If you use openai-generic, and your input has no images, we opt for using the simplified `string` type rather than the list of { type: "text" } ...
